### ISSUE TITLE
Dynamic look ahead

### ISSUE THREAD

**Issue by sonora** [osmandapp/OsmAnd#17318]
First suggested by @jsgblom, let me make sure this will not be overlooked without consideration:

We could introduce a new, fourth option for where your position is placed on the screen, called "Dynamic look ahead".

It would dynamically place the position icon on a centered ellipse.
-  The shape of the ellipse would be derived from your display's aspect ratio.
- The size of the ellipse will increase from zero with increasing speed.
- The size will also make sure the position icon is never covered by any widgets OsmAnd displays.
- The  placement of the position icon on the ellipse will be "backward" (with respect to your direction of movement) from the center, to provide more "look ahead" map real estate.

I have never seen this in action, but potentially it is a novel and useful feature for users in "North is up" Map orientation over the static "Center" placement?  Where e.g. smaller devices in portrait mode today have little look ahead when moving east or west? Or when moving north-east with many widgets displayed in the right column?

**Comment by jsgblom** [comment]
Off course, I'm in favour!

- I don't think the size of the ellipse should increase with speed. This makes it more complex to implent and potentially confusing for the user. 

- When not moving – speed ≈< 6km/h – the compass could be used. (Side note: I really miss the speed threshold compass / movement direction). Or just let it kick in when speed >≈ 6km/h 

I've been using it for quite some years now with the iOS app Pocket Earth. (I suggested it back then and they implemented it nicely.) I attached some screen shots. Also, note the arrow red/green, indicating portside/starboard which is handy as a reference and makes the position-arrow stand out visually.

![tempImagekBtcrJ](https://github.com/osmandapp/OsmAnd/assets/122881440/e830a619-da78-40f7-9cc1-6485cbb9bb07)

**Comment by vshcherb** [comment]
It sounds interesting but animation between changing orientation is not smooth. So if you change direction 180 degree map will start moving and same time rotating around the ellipse.

**Comment by vshcherb** [comment]
Also there is an issue what's the center of the screen cause it looks like center is related to my location bearing but if you pan the map you don't have the center connected to my location and it becomes very tricky where to open context menu what other actions (widget) related to the center will do

**Comment by sonora** [comment]
Yes, animation could be a problem, I think. Particularly when you start/stop moving, and/ or we cut over from using movement to compass as basis. (That's why I thought we may try to scale the ellipse size with speed. Not sure it's a valid idea.)

The "center of map" question less may be easier, because I think this mechanism would likely be applied only
- when the map is linled to the location
- and if the user selects the "Dynamic" option we may perhaps assume that the center of the ellipse is meant to be the screen center - I suggest it is not very useful to be in "bottom" position.

**Comment by jsgblom** [comment]
To see how Look Ahead in North-Up functions, have a look at Pocket Earth on an iPhone. Install the free version and go to _Settings » GPS & Compass_:

- Set _Map Rotation_ and _GPS Arrow_ to _Hybrid_. 
- Set _GPS Position_ to _Look Ahead_. 

_Hybrid_ means it will use compass heading when going slow (≈ 5km/u) and Course (moving direction) when going faster than ≈ 5km/u. (They use a window and Kalman filter to switch between the two so it is not switching all the time when going around 5km/u.)

In my opinion the "center of map" in North-Up should always be the center of the screen. BTW: when _Location position on screen_ is set to _Bottom_ it should also be in the center with North-Up. Now I can't use the _Bottom_ setting because it is really awkward to have the map-center at the bottom in North-Up. (Same line of thinking goes for the 2.5D view, which is only useful when you are rotating the map in the direction you are going – compass heading, coarse or manually.)

![IMG_4905](https://github.com/osmandapp/OsmAnd/assets/122881440/426daa73-dbc8-4853-ac93-766b1f383add)

**Comment by ossilator** [comment]
this sounds a lot like #6307.
